server/src/main/java/org/elasticsearch/common/time/CompoundDateTimeFormatter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.time;
+
+import org.elasticsearch.common.util.ArrayUtils;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+
+/**
+ * wrapper class around java.time.DateTimeFormatter that supports multiple formats for easier parsing,
+ * and one specific format for printing
+ */
+public class CompoundDateTimeFormatter {
+
+    final DateTimeFormatter printer;
+    final DateTimeFormatter[] parsers;
+
+    CompoundDateTimeFormatter(DateTimeFormatter parserAndPrinter, DateTimeFormatter ... parsers) {
+        if (parserAndPrinter == null) {
```

Comment:
This name seems deceptive because if you pass any parsers, then this is *not* used as a parser, right? If it should also always be a parser, then I think it would be better to just have a `formatters` member, and use `formatters[0]` for printing.

Reply:
indeed, way better. I'll push once CI passes... thx for reviewing!